kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -477,6 +500,25 @@ private void maybeBackOffEnqueuingNewEventWithKey(final KIn key) {
     }
   }
 
+  /**
+   * Invoked by the punctuator for this processor
+   */
+  private void punctuate(final long timestamp) {
```

Comment:
There's nothing new here, we just used to call `executeAvailableEvents` directly and now we have a dedicated method for punctuation (with error handling and logging now, so I guess that part is new)